lib/improver/tests/blending/weights/test_ChooseWeightsLinearFromDict.py

```diff
@@ -0,0 +1,513 @@
+            weighting_coord_name, config_coord_name, config_dict,
+            weights_coord_name=weights_coord_name)
+        result = str(plugin)
+        expected = ("<ChooseWeightsLinearFromDict "
+                    "config_dict = {}, "
+                    "weighting_coord_name = forecast_period, "
+                    "config_coord_name = model_configuration, "
+                    "weights_coord_name = alternative_name>".format(
+                        str(config_dict)))
+        self.assertEqual(result, expected)
+
+
+class Test__check_config_dict(IrisTest):
+    """Test the _check_config_dict method. """
+
+    def test_dictionary_key_mismatch(self):
+        """Test whether there is a mismatch in the dictionary keys. As
+        _check_config_dict is called within the initialisation,
+        _check_config_dict is not called directly."""
+        config_dict = {"uk_det": {"forecast_period": [7, 12, 48, 54],
+                                  "weights": [0, 1, 0],
+                                  "units": "hours"}}
+        weighting_coord_name = "forecast_period"
+        config_coord_name = "model_configuration"
+
+        msg = ('These items in the configuration dictionary')
+        with self.assertRaisesRegex(ValueError, msg):
+            ChooseWeightsLinearFromDict(
+                weighting_coord_name, config_coord_name,
+                config_dict=config_dict)
+
+    def test_dictionary_key_match(self):
+        """Test whether that the dictionary keys match in length as
+        expected."""
+        config_dict = {"uk_det": {"forecast_period": [7, 12, 48, 54],
+                                  "weights": [0, 1, 1, 0],
+                                  "units": "hours"}}
+        weighting_coord_name = "forecast_period"
+        config_coord_name = "model_configuration"
+        plugin = ChooseWeightsLinearFromDict(
+            weighting_coord_name, config_coord_name, config_dict=config_dict)
+        result = plugin._check_config_dict()
+        self.assertIsNone(result)
+
+
+class Test__arrange_interpolation_inputs(IrisTest):
+    """Test the _arrange_interpolation_inputs method. """
+
+    def test_basic(self):
+        """Test that the values for the source_points, target_points,
+        source_weights, axis and fill_value are as expected."""
+        expected_source_points = [7, 12, 48, 54]
+        expected_target_points = [6., 7., 8.]
+        expected_source_weights = [0, 1, 1, 0]
+        expected_fill_value = (0, 0)
+
+        weighting_coord_name = "forecast_period"
+        config_coord_name = "model_configuration"
+        config_dict = {"uk_det": {"forecast_period": [7, 12, 48, 54],
+                                  "weights": [0, 1, 1, 0],
+                                  "units": "hours"}}
+
+        cube = set_up_basic_model_config_cube()
+
+        plugin = ChooseWeightsLinearFromDict(
+            weighting_coord_name, config_coord_name, config_dict=config_dict)
```

Comment:
config_dict is not a keyword argument, so this can be slightly simpler.

Reply:
Corrected.